datasource/metadata/ec2/metadata.go

```diff
@@ -78,7 +78,11 @@ func (ms metadataService) FetchMetadata() (datasource.Metadata, error) {
 	}
 
 	if hostname, err := ms.fetchAttribute(fmt.Sprintf("%s/hostname", ms.MetadataUrl())); err == nil {
-		metadata.Hostname = strings.Split(hostname, " ")[0]
+		hostname := strings.Split(hostname, ".")[0]
+		if len(hostname) > 63 {
+			hostname = hostname[:63]
+		}
```

Comment:
do we need this truncation?

Reply:
yes please. I prefer either
a) no truncation here
b) truncate but warn that truncating

Either way the user will need to fix their hostname in their metadata service, and if we don't silently truncate it will be easier to track down.

Reply:
And we need to fix the test: https://github.com/flatcar/coreos-cloudinit/actions/runs/4931234078/jobs/8819414899?pr=19#step:5:21
Does it make sense to split by `" "` first and then "`.`"? Or is the test simply dumb.

Reply:
I am not sure the metadata service will ever return more than one. But I can split by space first and then by dot. One extra split won't hurt.

Will fix the test as well and push an update later today.